src/OrleansCodeGenerator/GrainMethodInvokerGenerator.cs

```diff
@@ -255,7 +255,7 @@ private static MethodDeclarationSyntax GenerateInvokeMethod(Type grainType, Meth
             for (var i = 0; i < methodParameters.Count; i++)
             {
                 var parameter = methodParameters[i];
-                var parameterType = parameter.ParameterType.GetTypeSyntax();
+                var parameterType = parameter.ParameterType.GetTypeSyntaxAsDynamic();
```

Comment:
I think you need to pass either `method`, `method.IsGenericMethod`, or the `ParameterInfo` itself to this modified `GetTypeSyntax` implementation, that way you know how to handle cases like:
```C#
interface IGenericGrain<T> : IGrainWithGuidKey
{
  Task<U> GenericMethod<U>(T t, U u);
}
```
In that case, you will have a bound generic type, `T`, and an unbound generic type `U` - you only want the latter to be `dynamic`, I believe.

If you go the `ParameterInfo` route, you can get the method from `(MethodInfo)info.Member` and then get the `DeclaringType` from the method so that you can determine whether or not the generic parameter is from a generic method or a generic interface.

Apologies if that's confusing. Feel free to PM me from the Gitter room if you want to talk about it.